File: src/test/programmers/python/level2/MoreSpicy.py
import heapq

def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville)

    while scoville[0] < K:
        heapq.heappush(scoville, heapq.heappop(scoville) + (heapq.heappop(scoville) * 2))
        answer += 1
        if len(scoville) == 1 and scoville[0] < K:
            answer = -1
            break
    return answer

print(solution([1, 2, 3, 9, 10, 12], 7))

File: src/test/programmers/java/level2/MoreSpicy.java
package test.programmers.java.level2;

import java.util.PriorityQueue;

public class MoreSpicy {
    public static void main(String[] args) {

    }

    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> queue = new PriorityQueue<Integer>();
        for(int i : scoville) {
            queue.offer(i);
        }

        while(queue.peek() < K) {
            queue.offer(queue.poll() + (queue.poll() * 2));
            answer++;

            if(queue.size() == 1 && queue.peek() < K) {
                answer = -1;
                break;
            }
        }

        return answer;
    }
}
